src/agent/coverage/src/cobertura.rs

```diff
@@ -1,12 +1,56 @@
 use crate::source::SourceCoverage;
-use crate::source::SourceCoverageLocation;
 use crate::source::SourceFileCoverage;
 use anyhow::Context;
 use anyhow::Error;
 use anyhow::Result;
+use std::path::Path;
 use std::time::{SystemTime, UNIX_EPOCH};
 use xml::writer::{EmitterConfig, XmlEvent};
 
+pub fn compute_line_rate(line_values: Vec<u32>) -> f32 {
+    let mut line_rate = 0_f32;
+    if line_values[1] > 0 {
+        line_rate = line_values[1] as f32 / line_values[0] as f32;
+    }
+    line_rate
+}
+
+pub fn compute_line_values_coverage(files: Vec<SourceFileCoverage>) -> (Vec<u32>, f32) {
```

Comment:
Apply earlier comment(3) about not passing owned values unless we need to.

Here, we probably want `files: &[SourceFileCoverage]`.

Reply:
Changed to pass in slice rather than vector.